crates/turborepo-auth/src/lib.rs

```diff
@@ -31,17 +38,143 @@ impl Token {
     pub fn existing(token: String) -> Self {
         Self::Existing(token)
     }
-    /// Checks if the token is valid. We do a few checks:
+    /// Checks if the token is still valid. The checks ran are:
+    /// 1. If the token is active.
+    /// 2. If the token has access to the cache.
+    ///     - If the token is forbidden from accessing the cache, we consider it
+    ///       invalid.
+    /// 3. We are able to fetch the user associated with the token.
+    ///
+    /// ## Arguments
+    /// * `client` - The client to use for API calls.
+    /// * `valid_message_fn` - An optional callback that gets called if the
+    ///   token is valid. It will be passed the user's email.
+    // TODO(voz): This should do a `get_user` or `get_teams` instead of the caller
+    // doing it. The reason we don't do it here is becuase the caller
+    // needs to do printing and requires the user struct, which we don't want to
+    // return here.
+    pub async fn is_valid<T: Client + TokenClient + CacheClient>(
+        &self,
+        client: &T,
+        // Making this optional since there are cases where we don't want to do anything after
+        // validation.
+        // A callback that gets called if the token is valid. This will be
+        // passed in a user's email if the token is valid.
+        valid_message_fn: Option<impl FnOnce(&str)>,
+    ) -> Result<bool, Error> {
+        let is_active = self.is_active(client).await?;
+        let has_cache_access = self.has_cache_access(client, None).await?;
+        if !is_active || !has_cache_access {
```

Comment:
I think we can just do these at the same time
```suggestion
        let (is_active, has_cache_access) = tokio::try_join!(self.is_active(client), self.has_cache_access(client, None))?;
        if !is_active || !has_cache_access {
```